src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesArray = [];
+
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+
+      case 'addProperties':
+        currentState = { ...currentState, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        const keys = Object.keys(currentState);
+        const keysToRemove = keys.filter((key) => action.keysToRemove.includes(key));
+
+        const newState = {};
+
+        keys.forEach((key) => {
+          if (!keysToRemove.includes(key)) {
+            newState[key] = currentState[key];
+          }
+        });
+
+        currentState = { ...newState }
+        break;
+
+      default:
```

Comment:
The default case in the switch statement is empty. While it's good to have a default case for error handling, it should at least provide some feedback, such as throwing an error or logging a message when an unknown action type is encountered.